lib/matplotlib/font_manager.py

```diff
@@ -1354,7 +1349,110 @@ def get_font_names(self):
         """Return the list of available fonts."""
         return list(set([font.name for font in self.ttflist]))
 
-    @lru_cache()
+    def _find_fonts_by_props(self, prop, fontext='ttf', directory=None,
+                             fallback_to_default=True, rebuild_if_missing=True):
+        """
+        Find font families that most closely match the given properties.
+
+        Parameters
+        ----------
+        prop : str or `~matplotlib.font_manager.FontProperties`
+            The font properties to search for. This can be either a
+            `.FontProperties` object or a string defining a
+            `fontconfig patterns`_.
+
+        fontext : {'ttf', 'afm'}, default: 'ttf'
+            The extension of the font file:
+
+            - 'ttf': TrueType and OpenType fonts (.ttf, .ttc, .otf)
+            - 'afm': Adobe Font Metrics (.afm)
+
+        directory : str, optional
+            If given, only search this directory and its subdirectories.
+
+        fallback_to_default : bool
+            If True, will fallback to the default font family (usually
+            "DejaVu Sans" or "Helvetica") if none of the families were found.
+
+        rebuild_if_missing : bool
+            Whether to rebuild the font cache and search again if the first
+            match appears to point to a nonexisting font (i.e., the font cache
+            contains outdated entries).
+
+        Returns
+        -------
+        list[str]
+            The paths of the fonts found
+
+        Notes
+        -----
+        This is an extension/wrapper of the original findfont API, which only
+        returns a single font for given font properties. Instead, this API
+        returns an dict containing multiple fonts and their filepaths
```

Comment:
a dict -> a list.
Perhaps make it explicit that not all fonts in `prop` may have a corresponding entry in the list (missing fonts are dropped, and at least one font is returned) (this may have been the original reason to return a dict, but I think returning a list and losing the 1-1 correspondence is just fine, as you still have the old findfont API available anyways).

Reply:
did go to a list.